github-runner-manager/tests/integration/test_debug_ssh.py

```diff
@@ -0,0 +1,499 @@
+
+    Yields:
+        A running application instance.
+    """
+    config = create_default_config(
+        github_config=github_config,
+        openstack_config=openstack_config,
+        proxy_config=proxy_config,
+        test_config=test_config,
+        ssh_debug_connections=[
+            SSHDebugConfiguration(
+                host=tmate_ssh_server.host,
+                port=tmate_ssh_server.port,
+                rsa_fingerprint=tmate_ssh_server.rsa_fingerprint,
+                ed25519_fingerprint=tmate_ssh_server.ed25519_fingerprint,
+                use_runner_http_proxy=False,
+                local_proxy_host="127.0.0.1",
+                local_proxy_port=3129,
+            )
+        ],
+    )
+    config_path = tmp_test_dir / "config.yaml"
+    config_path.write_text(yaml.dump(config), encoding="utf-8")
+
+    logger.info(
+        "Starting application with SSH debug configuration (test_id: %s)",
+        test_config.test_id,
+    )
+    metrics_log_path = tmp_test_dir / "github-runner-metrics.log"
+    log_file_path = test_config.debug_log_dir / f"application-{test_config.test_id}.log"
+    app = RunningApplication.create(
+        config_path, metrics_log_path=metrics_log_path, log_file_path=log_file_path
+    )
+
+    # Wait for runner to be created and setup reverse SSH tunnel
+    logger.info("Waiting for OpenStack runner to be created...")
+    runner, ip = wait_for_runner(openstack_connection, test_config)
+
+    if not runner or not ip:
+        pytest.fail("Failed to find OpenStack runner within timeout")
+
+    if not setup_reverse_ssh_tunnel(runner, ip, tmate_ssh_server):
+        pytest.fail("Failed to setup reverse SSH tunnel to tmate server")
+
+    yield app
+
+    logger.info("Stopping application")
+    app.stop()
+
+
+@pytest.mark.usefixtures("application_with_tmate_ssh_server")
+def test_tmate_ssh_connection(
+    test_config: TestConfig,
+    github_repository: Repository,
+    github_branch: Branch,
+    tmate_ssh_server: TmateServer,
+):
+    """Test that a tmate SSH connection can be established via the runner manager.
+
+    Arrange: Application configured with tmate SSH server connection details.
+    Act: Dispatch workflow that connects to the tmate SSH server.
+    Assert: Workflow completes successfully and logs contain server connection details.
+
+    Args:
+        test_config: Test-specific configuration for unique identification.
+        github_repository: Fixture providing the GitHub repository object.
+        github_branch: Fixture providing the GitHub branch object.
+        tmate_ssh_server: Fixture providing the tmate SSH server connection info.
+    """
+    dispatch_time = datetime.now(timezone.utc)
+    workflow = dispatch_workflow(
+        repository=github_repository,
+        workflow_filename=SSH_DEBUG_WORKFLOW_FILE_NAME,
+        ref=github_branch,
+        inputs={"runner": test_config.labels[0]},
+    )
+    workflow_run = get_workflow_dispatch_run(
```

Comment:
could we add a comment stating that this workflow will fail (similar as before in the deleted file ` # expect failure since the ssh workflow will timeout`).